### ISSUE TITLE
[Bug report] Cycle-free navigation in tmux panes could not be enabled?

### ISSUE THREAD

**Issue by kohane27** [aserowy/tmux.nvim#72]
Hello there. Hope you're doing well. Thank you for creating this unifying plugin; it makes switching between nvim buffers and tmux panes less cognitively demanding.

Please see the following video:

https://user-images.githubusercontent.com/57322459/175872082-8e279698-d0e4-42b5-98ba-3414ce510960.mp4

As you can see, pressing `Alt-h` cycles the tmux panes.

Inside my `.tmux.conf`:

```bash
# navigation
is_vim="ps -o state= -o comm= -t '#{pane_tty}' | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?(view|n?vim?x?)(diff)?$'"

bind-key -n 'M-h' if-shell "$is_vim" 'send-keys M-h' { if -F '#{pane_at_left}' '' 'select-pane -L' }
bind-key -n 'M-j' if-shell "$is_vim" 'send-keys M-j' { if -F '#{pane_at_bottom}' '' 'select-pane -D' }
bind-key -n 'M-k' if-shell "$is_vim" 'send-keys M-k' { if -F '#{pane_at_top}' '' 'select-pane -U' }
bind-key -n 'M-l' if-shell "$is_vim" 'send-keys M-l' { if -F '#{pane_at_right}' '' 'select-pane -R' }

bind-key -T copy-mode-vi 'M-h' if -F '#{pane_at_left}' '' 'select-pane -L'
bind-key -T copy-mode-vi 'M-j' if -F '#{pane_at_bottom}' '' 'select-pane -D'
bind-key -T copy-mode-vi 'M-k' if -F '#{pane_at_top}' '' 'select-pane -U'
bind-key -T copy-mode-vi 'M-l' if -F '#{pane_at_right}' '' 'select-pane -R'
```

`tmux.lua`:

```lua
require("tmux").setup({
  navigation = {
    -- C-hjkl
    enable_default_keybindings = false,
    cycle_navigation = false,
  },
})

local function map(mode, lhs, rhs, opts)
  local options = { noremap = true, silent = true }
  if opts then
    options = vim.tbl_extend("force", options, opts)
  end
  vim.api.nvim_set_keymap(mode, lhs, rhs, options)
end

map("n", "<A-h>", '<cmd>lua require("tmux").move_left()<cr>')
map("n", "<A-l>", '<cmd>lua require("tmux").move_right()<cr>')
map("n", "<A-k>", '<cmd>lua require("tmux").move_top()<cr>')
map("n", "<A-j>", '<cmd>lua require("tmux").move_bottom()<cr>')
```

As per instructed, "bindings in nvim must match the defined bindings in tmux". In nvim buffers, cycling navigation is indeed disabled:

https://user-images.githubusercontent.com/57322459/175873183-d543c671-d28c-4cc6-8888-1f7eadbd46c1.mp4

My question is, what went wrong and how do I disable cycling navigation in tmux panes as well?

Any input is much appreciated. Thank you:)

**Comment by aserowy** [comment]
Heho,

the config seems fine. Currently im not using tmux in my workflow anymore. Thus, i have no idea why it stopped working :D
To get a better grasp: What tmux version are you using? Maybe we have an issue on this end, because #{pane_at_left} should resolve only if there is a pane at the left.

Kinde regards
Alexander

**Comment by Normanras** [comment]
@aserowy Let me see if I can help with this and/or take it over. Sucks that you're not using tmux anymore!

@kohane27 can you please run in vim/neovim `:nmap <A-h>` (along with the other keys) and let me know what command those keys are mapped to?

Have you installed any other plugins recently?

**Comment by kohane27** [comment]
@Normanras 

Thank you for taking a look into this. I appreciate it:)

```
:nmap <A-h>
n  <M-h>       * <Cmd>lua require("tmux").move_left()<CR>

:nmap <A-l>
n  <M-h>       * <Cmd>lua require("tmux").move_right()<CR>

:nmap <A-j>
n  <M-h>       * <Cmd>lua require("tmux").move_bottom()<CR>

:nmap <A-k>
n  <M-h>       * <Cmd>lua require("tmux").move_top()<CR>
```

The problem is not within neovim; it's that I can't disable pane cycling within two terminals. Pane cycling is disabled in two neovim buffers.

I haven't installed any other plugins recently.

Thank you again!

**Comment by aserowy** [comment]
@Normanras 

tyvm!

I mean, i am not using tmux anymore because i switched my working environment to linux completly :) i would say this is a good thing :D

**Comment by kohane27** [comment]
Hello guys! I figured it out.

TL;DR: it was a keybinding conflict with a tmux plugin I use called (https://github.com/jabirali/tmux-tilish)[tmux-tilish].

The following is helpful in helping you debug if you encounter similar problems.

If you achieve cycle-free navigation in nvim, it means it's not a problem with `tmux.nvim`. The problem very likly lies within your tmux config.

Try the default keybindings outlined in [tmux.nvim#navigation](https://github.com/aserowy/tmux.nvim#navigation). Do not remap just yet. Remember to `tmux kill-sever` to make sure `tmux` reloads the changes. If it works, then it means you have keybinding conflicts within your tmux environment. This was my case.

If you happened to use (https://github.com/jabirali/tmux-tilish)[tmux-tilish], go to its source code and comment out the following.
```bash
# Switch to pane via Alt + hjkl.
tmux $bind "${mod}${h}" select-pane -L
tmux $bind "${mod}${j}" select-pane -D
tmux $bind "${mod}${k}" select-pane -U
tmux $bind "${mod}${l}" select-pane -R
```

You also need to set `set -g @tilish-navigator 'off'` and cycle-free navigation in your `.tmux.conf`. The full code:

```bash
set -g @tilish-navigator 'off'

is_vim="ps -o state= -o comm= -t '#{pane_tty}' | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?(view|n?vim?x?)(diff)?$'"

# navigation
bind-key -n 'M-h' if-shell "$is_vim" 'send-keys M-h' { if -F '#{pane_at_left}' '' 'select-pane -L' }
bind-key -n 'M-j' if-shell "$is_vim" 'send-keys M-j' { if -F '#{pane_at_bottom}' '' 'select-pane -D' }
bind-key -n 'M-k' if-shell "$is_vim" 'send-keys M-k' { if -F '#{pane_at_top}' '' 'select-pane -U' }
bind-key -n 'M-l' if-shell "$is_vim" 'send-keys M-l' { if -F '#{pane_at_right}' '' 'select-pane -R' }

bind-key -T copy-mode-vi 'M-h' if -F '#{pane_at_left}' '' 'select-pane -L'
bind-key -T copy-mode-vi 'M-j' if -F '#{pane_at_bottom}' '' 'select-pane -D'
bind-key -T copy-mode-vi 'M-k' if -F '#{pane_at_top}' '' 'select-pane -U'
bind-key -T copy-mode-vi 'M-l' if -F '#{pane_at_right}' '' 'select-pane -R'
```

Reload and it should work. Cheers!
